sdk/tests/iot/hub/test_az_iot_hub_client_commands.c

```diff
@@ -0,0 +1,585 @@
+}
+
+static void test_az_iot_hub_client_commands_parse_received_topic_topic_filter_fail()
+{
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  options.model_id = test_model_id;
+
+  az_iot_hub_client client;
+  assert_true(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span received_topic = az_span_create(
+      g_expected_methods_subscribe_topic, _az_COUNTOF(g_expected_methods_subscribe_topic));
+
+  az_iot_hub_client_command_request out_request;
+
+  assert_true(
+      az_iot_hub_client_commands_parse_received_topic(&client, received_topic, &out_request)
+      == AZ_ERROR_IOT_TOPIC_NO_MATCH);
+}
+
+static void test_az_iot_hub_client_commands_parse_received_topic_response_topic_fail()
+{
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  options.model_id = test_model_id;
+
+  az_iot_hub_client client;
+  assert_true(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span received_topic = AZ_SPAN_FROM_STR("$iothub/methods/res/200/?$rid=2");
+
+  az_iot_hub_client_command_request out_request;
+
+  assert_true(
+      az_iot_hub_client_commands_parse_received_topic(&client, received_topic, &out_request)
+      == AZ_ERROR_IOT_TOPIC_NO_MATCH);
+}
+
+static const az_span _log_expected_topic
+    = AZ_SPAN_LITERAL_FROM_STR("$iothub/methods/POST/TestMethod/?$rid=1");
+static int _log_invoked_topic = 0;
+static void _log_listener(az_log_classification classification, az_span message)
+{
+  switch (classification)
+  {
+    case AZ_LOG_MQTT_RECEIVED_TOPIC:
+      assert_memory_equal(
+          az_span_ptr(_log_expected_topic), az_span_ptr(message), (size_t)az_span_size(message));
+      _log_invoked_topic++;
+      break;
+    default:
+      assert_true(false);
+  }
+}
+
+static void test_az_iot_hub_client_commands_logging_succeed()
+{
+  az_log_set_message_callback(_log_listener);
+
+  assert_int_equal(0, _log_invoked_topic);
+
+  _log_invoked_topic = 0;
+
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  options.model_id = test_model_id;
+
+  az_iot_hub_client client;
+  assert_true(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_iot_hub_client_command_request out_request;
+  assert_true(
+      az_iot_hub_client_commands_parse_received_topic(&client, _log_expected_topic, &out_request)
+      == AZ_OK);
+
+  assert_int_equal(_az_BUILT_WITH_LOGGING(1, 0), _log_invoked_topic);
+  // assert_false(true);
+
+  az_log_set_message_callback(NULL);
+}
+
+#ifdef _MSC_VER
+// warning C4113: 'void (__cdecl *)()' differs in parameter lists from 'CMUnitTestFunction'
+#pragma warning(disable : 4113)
+#endif
+
+/*static int teardown(void **state) {
+     (void)state;
+     printf("*** in teardown ***\n");
+     //az_log_set_message_callback(NULL);
+     return 0;
+}*/
```

Comment:
is this leftover from the previous tests or is this new? Why have it?

Reply:
No - embarassed this made it in, good catch.  We need something like this to reset bad log test runs, but... it needs to go into main repo in separate PR since all log tests have this.  We need to use cmocka_teardown or whatever ot put in the teardown callback.

Reply:
gotcha sounds good